lib/iris/tests/unit/plot/test_2d_coords.py

```diff
@@ -135,32 +114,120 @@ def test_map_common_2d(self):
         result = iplt._map_common('pcolor', None, self.mode, cube, plot_defn)
         self.assertTrue(result)
 
-#    def test_discontiguous_masked(self):
-#        # Test that a contiguity check will raise a warning (not an error) for
-#        # discontiguous bounds but appropriately masked data.
-#        cube = self.latlon_2d
-#        coord = cube.coord('longitude')
-#        msg = 'The bounds of the longitude coordinate are not contiguous.  ' \
-#              'However, data is masked where the discontiguity occurs so ' \
-#              'plotting anyway.'
-#        with self.assertWarnsRegexp(msg):
-#            iplt._check_contiguity_and_bounds(coord, cube.data)
+    def test_draw_2d_from_bounds(self):
+        # Test this function will not raise an error even with our most
+        # awkward but supported cube.
+        cube = self.latlon_2d
+        result = iplt._draw_2d_from_bounds('pcolormesh', cube)
+        self.assertTrue(result)
+
+
+class TestContiguityChecking(tests.IrisTest):
+    def setUp(self):
+        self.latlon_2d = full2d_global()
+        make_bounds_discontiguous_at_point(self.latlon_2d, 3, 2)
+        self.lons = self.latlon_2d.coord('longitude')
+
+        # Construct an array the size of the points array filled with 'False'
+        # to represent an empty discontiguity array
+        self.empty_bool = np.zeros(self.lons.points.shape,
+                                   dtype=bool)
+        self.masked = self.empty_bool.copy()
+        self.masked[3, 3] = True
 
     def test_discontiguous_unmasked(self):
-        # Check that an error occurs when the contiguity check finds
+        # TODO Decide whether to throw a warning or an error and alter this
+        # test accordingly
+
+        # Check that a warning occurs when the contiguity check finds
         # discontiguous bounds but unmasked data.
         cube = self.latlon_2d
         cube.data.mask = ma.nomask
+        coord = self.lons
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
+            with self.assertRaises(UserWarning):
```

Comment:
This really needs a `assertRaisesRegexp` to check it's getting the right warning`